transmute_core/tests/test_schematics.py

```diff
@@ -165,3 +165,27 @@ def test_can_handle(object_serializer_set, cls, should_handle):
     the object serializer set should not raise an exception if it can handle these types.
     """
     object_serializer_set[cls]
+
+
+def test_schematics_uses_cached_entries():
+    """
+    Regression test. Validating that SchematicsSerializer uses the
+    same json schema it used previously, as before erroneous
+    use of the instantiated model as a key was causing memory leaks.
+    """
+    serializer = SchematicsSerializer()
+    # A nested schema type is required as primitives have
+    # hard-coded dictionaries representing the json.
+    class SchematicsBody(Model):
+        name = StringType(max_length=5)
+
+    # ensure that instances have the same key as well.
+    instance = ModelType(SchematicsBody)
+    original_payload = serializer.to_json_schema(instance)
+    second_payload = serializer.to_json_schema(instance)
+    assert original_payload is second_payload
```

Comment:
Should we expect two different instances of the same thing to be the same payload? e.g.
```
    instance = ModelType(SchematicsBody)
    second_instance = ModelType(SchematicsBody)
    original_payload = serializer.to_json_schema(instance)
    second_payload = serializer.to_json_schema(second_instance)
    assert original_payload is second_payload
```

Reply:
as far as transmute is concerned, it would be a weird pattern to wrap your type in a ModelType in a non-nested scenario. But could definitely see people encountering it.

This won't cause a memory leak as ModelTypes are only created dynamically once when the raw type is passed through, and cached from that point on.

I think it's a good optimization, but if I was to clean this up properly I'd probably add more strict validation to reject weird edge cases like this. If you'd like to follow up with that optimization in a PR I encourage it, but I'm personally not sure the extra complexity is worth a few fewer cache entries.